doc/glossary.rst

```diff
@@ -0,0 +1,1350 @@
+
+        Semantically, a multilabel target can be thought of as a set of labels
+        for each sample.  While not used internally,
+        :class:`preprocessing.MultiLabelBinarizer` is provided as a utility to
+        convert from a list of sets representation to a 2d array or sparse
+        matrix. One-hot encoding a multiclass target with
+        :class:`preprocessing.LabelBinarizer` turns it into a multilabel
+        problem.
+
+        :func:`~utils.multiclass.type_of_target` will return
+        'multilabel-indicator' for multilabel input, whether sparse or dense.
+
+    multioutput continuous
+        TODO
+
+    multioutput multiclass
+        TODO
+
+        A classification problem...
+
+        :mod:`multioutput` provides estimators which estimate multi-output
+        problems using multiple single-output estimators.  This may not fully
+        account for dependencies among the different outputs, which methods
+        natively handling the multioutput case (e.g. decision trees, nearest
+        neighbors, neural networks) may do better.
+
+.. _glossary_methods:
+
+Methods
+=======
+
+.. glossary::
+
+    ``decision_function``
+        In a fitted :term:`classifier` or :term:`outlier detector`, predicts a
+        "soft" score for each sample in relation to each class, rather than the
+        "hard" categorical prediction produced by :term:`predict`.  Its input
+        is usually only some observed data, :term:`X`.
+
+        Output conventions:
+
+        binary classification
+            A 1-dimensional array, where values strictly greater than zero
+            indicate the positive class (i.e. the last class in
+            :term:`classes_`).
+        multiclass classification
+            A 2-dimensional array, where the row-wise arg-maximum is the
+            predicted class.  Columns are ordered according to
+            :term:`classes_`.
+        multilabel classification
+            Scikit-learn is inconsistent in its representation of multilabel
+            decision functions.  Some estimators represent it like multioutput
+            multiclass, i.e. a list of 2d arrays, each with two columns. Others
+            represent it with a single 2d array, whose columns correspond to
+            the individual binary classification decisions. The latter
+            representation is ambiguously identical to the multiclass
+            classification format, though its semantics differ: it should be
+            interpreted, like in the binary case, by thresholding at 0.
+
+            TODO
+            ..
+            see https://gist.github.com/jnothman/4807b1b0266613c20ba4d1f88d0f8cf5
+        multioutput classification
+            A list of 2d arrays, corresponding to each multiclass decision
+            function.
+        outlier detection
+            A 1-dimensional array, where a value greater than or equal to zero
+            (TODO: check equality case) indicates an inlier.
```

Comment:
with PR #9015 'greater than or equal to zero' will be correct for all outlier detectors so you can remove the TODO